### PR TITLE
Fixing a COM memory leak in the WMI wrapper

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit f83891c7c4e244bd51a3b0cdd62e7170516fa290 (Job results: [425](https://jenkins.osquery.io/job/osqueryPullRequestBuild/425/)) failed one or more tests (**Windows**).

**Comment by PoppySeedPlehzr** [comment]
While this looks to alleviate issues with the daemon, the shell no longer connects to the COM db. I'm investigating now.

**Comment by theopolis** [comment]
You will need to add the calls to the start/shutdown in test_utils too. The testing "main" does not use the initializer: https://github.com/facebook/osquery/blob/master/osquery/main/tests.cpp

Perhaps introduce new static methods to Initializer: platformSetUp and platformTearDown and let POSIX/Windows implement each? Then the initializer can call it in its ctor, and otherwise tests can call the static directly?

**Comment by PoppySeedPlehzr** [comment]
@theopolis what else do you see existing in the platformSetup/Teardown functions? I feel like the `std::signal` calls are all that I see that would be pulled 'up'.

Additionally, I may be refactoring this wrong, but with a new header and static definitions for the platformSetup/Teardown, it seems like I'll still need to call these things in the shell's `main`? Which is odd as there's an `Initializer` ctor there...

**Comment by osqueryer** [comment]
:no_good: The commit dbd78a85f77040a8a7a326960d30680021215bd1 (Job results: [405](https://jenkins.osquery.io/job/osqueryPullRequestAudit/405/)) failed the code audit.

File: osquery/core/init.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
```
Comment (theopolis) [inline osquery/core/init.h]: Delete this file, Initializer is already defined in system.

File: osquery/tests/test_util.cpp [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #include <osquery/logger.h>
 #include <osquery/sql.h>
 
+#include "osquery/core/init.h"
```
Comment (theopolis) [inline osquery/tests/test_util.cpp]: Remove this, swap for system.

File: osquery/tests/test_util.cpp [hunk 3]
```diff
@@ -101,10 +102,19 @@ void initTesting() {
   // Set up the database instance for the unittests.
   DatabasePlugin::setAllowOpen(true);
   DatabasePlugin::initPlugin();
+
+#ifdef WIN32
```
Comment (theopolis) [inline osquery/tests/test_util.cpp]: No ifdef needed.

File: osquery/tests/test_util.cpp [hunk 4]
```diff
@@ -101,10 +102,19 @@ void initTesting() {
...
+  // On Windows tests leverage WMI, so we init the COM libs here
+  osquery::platformSetup();
+#endif
 }
 
 void shutdownTesting() {
   DatabasePlugin::shutdown();
```
Comment (theopolis) [inline osquery/tests/test_util.cpp]: Same, and comments are duplicated.

**Comment by theopolis** [comment]
If this builds on Windows it should be good to go!

**Comment by PoppySeedPlehzr** [comment]
This mostly addresses #2575

**Pull request merged**